spec/index.md

```diff
@@ -598,8 +599,102 @@ MISSING
 
 ### 5.1.1.13. Event types
 
-An *event type* is the type denotated by the name of an [`event` item].
+An *event type* is the type denoted by the name of an [`event` item].
 
+## 6. Data Layout
+
+There are three places where data can be stored on the EVM:
+
+- stack
+- storage
+- memory
+
+Each data type described in section 5 can be stored in these locations. How data is stored is 
+described in this section.
+
+### 6.1. Stack
+
+The following can be stored on the stack:
+
+- base type values
+- pointers to sequence type values
+
+Sequence type values cannot be stored on the stack because they can exceed the stack slot size, so
+we use pointers stored on the stack that reference them.
+
+Example:
+
+```python
+# function scope
+foo: u256 = 42 # foo is stored on the stack
+bar: u256[100] # bar is a memory pointer stored on the stack
+```
+
+### 6.2. Storage
+
+The following can be stored in storage:
+
+- base type values
+- reference type values
+
+#### 6.2.1. Constant size values in storage
+
+Storage pointers for constant size values are determined at compile time. 
+
+Example:
+
+```python
+# contract scope
+foo: u256 # foo is assigned a static pointer by the compiler
+```
+
+The value of a base type in storage is found by simply loading the value from storage at the
+given pointer.
+
+To find a value inside of a sequence type. The relative location of the value is added to the
+given pointer.
+
+#### 6.2.2. Maps in storage
+
+Maps are not assigned pointers, because they do not have a location in storage. They are instead 
+assigned a unique index value that is used to derive the location of keyed values.
```

Comment:
What do you think about the term `nonce` instead of `index`?

https://www.merriam-webster.com/dictionary/nonce

It denotes *single use* but without the notion of ordering or *indexing* into a sequence.

Reply:
I was actually trying to think of a word other than index. Nonce makes alot of sense.